src/smt/solver_engine.cpp

```diff
@@ -1397,6 +1394,14 @@ std::vector<Node> SolverEngine::getAssertionsInternal() const
 
 const Options& SolverEngine::options() const { return d_env->getOptions(); }
 
+bool SolverEngine::isWellFormedTerm(const Node& n) const
+{
+  // Must rewrite before checking for free variables
+  Node nr = d_env->getRewriter()->rewrite(n);
```

Comment:
I am not in favor of this. The rewriter is quite complex. So, users will have trouble figuring out why some terms are accepted, while others are rejected.

Reply:
This is actually already how the other checks work: https://github.com/cvc5/cvc5/pull/11260/files#diff-22e2451898c88f4111e59d72affa7ec7b2fb8d07e00e2ab07d30d3170d2a5befR1411

I agree its not ideal.

Reply:
I think the solution is to make `hasFreeVar` agnostic to variable shadowing issues. I will look into this soon.

Reply:
I've updated this PR with this simplification.